qt/TorrentDelegate.cc

```diff
@@ -606,7 +611,39 @@ void TorrentDelegate::drawTorrent(QPainter* painter, QStyleOptionViewItem const&
     progress_bar_style_.state = progress_bar_state;
     setProgressBarPercentDone(option, tor);
 
-    StyleHelper::drawProgressBar(*style, *painter, progress_bar_style_);
+    auto const cr = is_item_selected ? QPalette::Highlight : QPalette::Base;
+    float const progress = static_cast<float>(progress_bar_style_.progress) / static_cast<float>(progress_bar_style_.maximum);
+
+    QRect progress_bar_fill_rect = layout.progress_bar_rect;
+    progress_bar_fill_rect.setWidth(progress * layout.progress_bar_rect.width());
+    painter->fillRect(progress_bar_fill_rect, progress_bar_style_.palette.brush(cr));
+
+    std::vector<int> const availability = tor.availability();
+    float const piece_width = static_cast<float>(layout.piece_bar_rect.width() - 3) / tor.pieceCount();
+
+    if (progress == 1.0)
+    {
+        painter->fillRect(layout.piece_bar_rect, progress_bar_style_.palette.brush(cr));
+    }
+    else
+    {
+        for (int i = 0; i < tor.pieceCount(); i++)
+        {
+            if (availability[i] == -1)
+            {
+                int const offset = i * piece_width;
+                QRect const piece_rect = { layout.piece_bar_rect.left() + offset + 2,
+                                           layout.piece_bar_rect.top() + 1,
+                                           static_cast<int>(std::ceil(piece_width)),
+                                           PIECE_BAR_HEIGHT - 3 };
+
+                painter->fillRect(piece_rect, progress_bar_style_.palette.brush(cr));
+            }
+        }
+    }
+
+    painter->drawRect(layout.progress_bar_rect);
```

Comment:
These calls seem to be using text color which may not be suitable (e.g. it'll be red in case error state, which may not go well right next to green or blue). And in fact, it looks too bright for Gimp on your screenshot. Consider using `QPalette::{Light,Midlight,Dark,Mid,Shadow}` instead.